### ISSUE TITLE
`no-unused-prop-types` doesn't check `nextProps` of `componentWillReceiveProps`

### ISSUE THREAD

**Issue by byronmwong** [yannickcr/eslint-plugin-react#801]
If a prop is only used in the context of `nextProps`, the lint rule mistakenly thinks that the prop is never used.

Here's a repro case:

```
import React, {PropTypes} from "react";

export default React.createClass({

  propTypes: {
    foo: PropTypes.bool, // this line gets flagged when it shouldn't
    bar: PropTypes.func,
  },

  componentWillReceiveProps (nextProps) {
    if (nextProps.foo) {
      return;
    }
  },

  render () {
    this.props.bar;
    return <div />;
  }
});
```

**Comment by ljharb** [comment]
What's the use case for a prop that's not used in the render path?

**Comment by byronmwong** [comment]
It might not be idiomatic React, but in this case, we're waiting on a prop
to turn `true` and when it does, we trigger a route change. You could
imagine a signup form component waiting for `nextProps.signupCompleted` to
be true, and then redirecting to the homepage.

**Comment by EvNaverniouk** [comment]
This will be fixed by https://github.com/yannickcr/eslint-plugin-react/pull/792

**Comment by byronmwong** [comment]
Great! Thanks @EvNaverniouk

**Comment by mjilugu** [comment]
Hi, I still get this same error. When we use a prop only with `nextProps` to set the state in the `componentWillRecieveProps(nextProps)` method and not elsewhere we get `react/no-used-prop-types` error for the prop in the propTypes block.

**Comment by wx2228** [comment]
still getting this error too.

**Comment by jseminck** [comment]
This should be fixed. @wx2228 what version of eslint-plugin-react do you use? If using the latest, could you provide sample code of what is failing? Then I can look into it. Thanks!

**Comment by wx2228** [comment]
@jseminck  thanks for the prompt reply. I am using 6.4.1. I will upgrade it

**Comment by andfelzapata** [comment]
it also wont check nextProps if componentWillReceiveProps is declared as an arrow function.

**Comment by ljharb** [comment]
@andfelzapata would you file that as a separate issue?
